### PR TITLE
Added the ability to set a custom request rate limit for Search streams

### PR DESCRIPTION
**Pull request by JamesFrost**
While the Search endpoint has a default rate limit, it can vary between contracts. As such, the ability to set custom rate limits is needed.

The default rate limit has been changed from 10/second to 30/minute, based on information given by our GNIP account manager.

This update is backwards compatible - if a rate limiter isn't specified, it defaults to the rate limit specified above.

Also included is a fix for Search streams returning a 422 'Invalid JSON' error due to a malformed POST request body.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -131,6 +131,19 @@ Used to segregate rules and their matching data into different logical groups. O
...
 
+#### options.rateLimiter
+A <a href="https://www.npmjs.com/package/limiter">limiter</a> object, used to control the rate of collection. Optional. If unspecified, a rate limit of 30 requests a minute will be shared between Search streams. If you have a non-standard rate limit, you should pass this parameter.
+
+```js
+var RateLimiter = require('limiter').RateLimiter;
+// Allow 60 requests per minute
+var limiter = new RateLimiter(60, 'minute');
+
+var stream = new _gnip.Search({
```
Comment (demian85) [inline README.md]: Is `_gnip` a typo?

File: lib/index.js [hunk 2]
```diff
@@ -156,6 +156,7 @@ exports.Stream = GnipStream;
 exports.Rules = GnipRules;
 exports.Usage = GnipUsage;
 
-exports.Search = function (options) {
-  return new GnipSearch(options, searchApiRateLimiter);
+exports.Search = function( options ) {
+	rateLimiter = options.rateLimiter || searchApiRateLimiter;
```
Comment (demian85) [inline lib/index.js]: global var? :/

File: lib/search.js [hunk 3]
```diff
@@ -115,20 +115,18 @@ GnipSearch.prototype._makeRequest = function (next, callback) {
...
-    body: requestData
-  }, callback);
+    const auth = "Basic " + new Buffer(this.options.user + ":" + this.options.password).toString("base64");
+
+	request({
+	    url: this.options.url,
+	    method: "POST",
+	    json: true,
+	    headers: {
+            "Authorization" : auth,
```
Comment (demian85) [inline lib/search.js]: can you please indent this line properly?

**Comment by demian85** [comment]
Please review my comments so I can merge.

**Comment by demian85** [comment]
If you have a Gnip account, can you please run `test.js` to verify?
Improvements to the test file are welcome also!
Thanks.

**Comment by JamesFrost** [comment]
@demian85 I've made the requested changes, and the test file ran correctly 👍

**Pull request merged**